analysis_functions.py

```diff
@@ -0,0 +1,160 @@
+import numpy as np
+import polars as pl
+from matplotlib import pyplot as plt
+from scipy import stats
+
+def get_general_ch_statistics(filtered_df, cols_to_keep = ['dislike_count','duration','like_count','view_count'], channel = False):
+
+    '''get general statistics for all channel on a given column'''
```

Comment:
Add a description of the arguments for your functions. Something like:
```python
def get_general_ch_statistics(filtered_df, cols_to_keep = ['dislike_count','duration','like_count','view_count'], channel = False):

    '''get general statistics for all channel on a given column
      Args:
      - filtered_df: The dataframe containing the channel information.
      - cols_to_keep: The columns the analysis is performed on. Default: ['dislike_count','duration','like_count','view_count']
      - channel: If false, the name of the channel column is 'channel_id' else it is 'channel'
      Returns:
      - counts,means,stdevs,medians: the .... of the channel
'''
```
The same applies for the other functions.